### PR TITLE
chore: remove unused execa devDependencies

### PR DESCRIPTION
**Pull request by benmccann**
### Description

Removes a few usages of execa from the project to make an eventual transition to `tinyexec` easier

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Comment by sheremet-va** [comment]
Any reason why? We don't want to upgrade execa because `@antfu/install-pkg` in our pinned version relies on the same version so updating it will cause two execas to be bundled, and we are waiting on using [`tinyexec`](https://github.com/tinylibs/tinyexec):  https://github.com/vitest-dev/vitest/pull/5713

File: packages/vitest/src/node/git.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export class VitestGit {
   constructor(private cwd: string) {}
 
   private async resolveFilesWithGitCommand(args: string[]): Promise<string[]> {
-    let result: ExecaReturnValue
+    let result: Result
```
Comment (AriPerkkio) [inline packages/vitest/src/node/git.ts:15]: This should narrow doens the type of `result.stdout` below to `string|undefined`:

```suggestion
    let result: Result<{}>
```

**Comment by benmccann** [comment]
Ok, great! Thanks for those added details. I've reverted the upgrade, but left this PR open with the other commit it contained which removes a few unused instances of `execa`. I've update the title and description of the PR accordingly. This PR should make the eventual planned transitions to `ezspawn` or `tinyexec` easier

**Comment by benmccann** [comment]
Hmm. It's weird that the CI is failing. It was passing with my prior commit. It's complaining about declining test coverage for a vue test and I haven't touched that. This looks like a problem with the CI setup rather than the PR

**Comment by AriPerkkio** [comment]
Not sure what's going on there. The error itself is not related to coverage at all. All `expect` assertions in this test fixture failed. 

https://github.com/vitest-dev/vitest/blob/eae3bf4a72c1ab0b0b8ea0e13a3ed8a415ebfc66/test/coverage-test/fixtures/test/vue-fixture.test.ts#L1-L52

I haven't seen these failures before so cannot say that it's flaky - especially now that all 4 CI machines with different OS/Node failed at the same time. 🤔

**Comment by benmccann** [comment]
Thanks! It seems not to be flaky, but also not related to this PR. I opened another PR with an unrelated change and it is also failing in the same manner: https://github.com/vitest-dev/vitest/pull/6298

**Comment by AriPerkkio** [comment]
I think the `main` CI broke somewhere during here: https://github.com/vitest-dev/vitest/commits/main/?since=2024-08-03&until=2024-08-08. Some merges have skipped the CI I guess, cc. @sheremet-va.

**Pull request merged**